### PR TITLE
Feature / Rename API tests

### PR DESCRIPTION
**Pull request by Jerry-Jinfeng-Guo**
This PR renames API tests to drop the "c_" and "cpp_" prefix to be more concise.

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@ find_package(doctest REQUIRED)
 
 include("${doctest_DIR}/doctest.cmake")
 
-add_subdirectory("c_api_tests")
+add_subdirectory("api_tests")
```
Comment (mgovers) [inline tests/CMakeLists.txt:9]: This is ambiguous because itbdoesbnotbtest the python API

**Review by mgovers — COMMENTED** [review]
Otherwise looks OK

File: tests/CMakeLists.txt (hunk 1 above)
Comment (Jerry-Jinfeng-Guo) [reply tests/CMakeLists.txt:9]: `native_api_tests`?

Comment (mgovers) [reply tests/CMakeLists.txt:9]: yes, that's much better indeed

Comment (Jerry-Jinfeng-Guo) [reply tests/CMakeLists.txt]: It's now in the latest commit

**Review by figueroa1395 — APPROVED** [review]
Makes sense and looks good to me. However, just to double check, is there anything else to it besides making the naming more consistent? If no, feel free to merge.

**Comment by Jerry-Jinfeng-Guo** [comment]
> Makes sense and looks good to me. However, just to double check, is there anything else to it besides making the naming more consistent? If no, feel free to merge.

Nothing else indeed.

**Pull request merged**